nemo/collections/nlp/models/token_classification/ner_model.py

```diff
@@ -0,0 +1,215 @@
+
+from nemo.collections.common.losses import CrossEntropyLoss
+from nemo.collections.common.tokenizers.bert_tokenizer import NemoBertTokenizer
+from nemo.collections.nlp.data.token_classification.token_classification_dataset import BertTokenClassificationDataset
+from nemo.collections.nlp.data.token_classification.token_classification_descriptor import TokenClassificationDataDesc
+from nemo.collections.nlp.metrics.classification_report import ClassificationReport
+from nemo.collections.nlp.modules.common import TokenClassifier
+from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
+from nemo.core.classes import typecheck
+from nemo.core.classes.modelPT import ModelPT
+from nemo.core.neural_types import NeuralType
+from nemo.utils.decorators import experimental
+
+__all__ = ['NERModel']
+
+
+@experimental
+class NERModel(ModelPT):
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.bert_model.input_types
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.classifier.output_types
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        """
+        Initializes BERT Named Entity Recognition model.
+        """
+
+        self.data_desc = TokenClassificationDataDesc(
+            data_dir=cfg.data_dir, modes=["train", "test", "dev"], pad_label=cfg.train_ds.pad_label
+        )
+        self.data_dir = cfg.data_dir
+
+        # TODO add support for sentence_piece tokenizer
+        if cfg.language_model.tokenizer == 'nemobert':
+            self.tokenizer = NemoBertTokenizer(pretrained_model=cfg.language_model.pretrained_model_name)
+        else:
+            raise NotImplementedError()
+
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        self.bert_model = get_pretrained_lm_model(
+            pretrained_model_name=cfg.language_model.pretrained_model_name,
+            config_file=cfg.language_model.bert_config,
+            checkpoint_file=cfg.language_model.bert_checkpoint,
+        )
+        self.hidden_size = self.bert_model.config.hidden_size
+
+        self.classifier = TokenClassifier(
+            hidden_size=self.hidden_size,
+            num_classes=self.data_desc.num_classes,
+            num_layers=cfg.head.num_fc_layers,
+            activation=cfg.head.activation,
+            log_softmax=cfg.head.log_softmax,
+            dropout=cfg.head.fc_dropout,
+            use_transformer_init=cfg.head.use_transformer_init,
+        )
+
+        if cfg.class_balancing == 'weighted_loss':
+            # You may need to increase the number of epochs for convergence when using weighted_loss
+            self.loss = CrossEntropyLoss(weight=self.data_desc.class_weights)
+        else:
+            self.loss = CrossEntropyLoss()
+
+        # setup to track metrics
+        self.class_report = ClassificationReport(self.data_desc.num_classes, label_ids=self.data_desc.label_ids)
```

Comment:
Can we take the max of label_ids as num_classes to drop the first input?

Reply:
label_ids is optional for tasks that have int labels it won't be provided